assets/js/src/admin/popup-editor/general.js

```diff
@@ -56,6 +56,79 @@
 					}
 				}
 			});
+
+			document.querySelector( '#pum-popup-settings-container' ).addEventListener( 'click', function(e) {
+				if ( Array.from(e.target.classList).includes('popup-type') || Array.from(e.target.parentElement.classList).includes('popup-type') ) {
+					const $container = jQuery( '#pum-popup-settings-container' );
+					if ( 1 === $container.length ) {
+						// Temporarily hardcoded object for testing...
+						const popupTypes = {
+							'center-popup': {
+								'size': 'medium',
+								'responsive_min_width': '0%',
+								'responsive_max_width': '100%',
+								'animation_type': 'fade',
+								'animation_speed': 350,
+								'location': 'center',
```

Comment:
Change this to top center, add top_position = 20 or 40px whichever you prefer. Setting it to center doesn't work well on mobile by default if the content is taller than the screen. They can't close it.

Reply:
Since so many people ask about how to center the popup, the goal of this one was to have that as a quick preset. Maybe we break this into two different presets. One for the full takeover and one for a center popup.

Reply:
Not sure where you are seeing this request. Nobody I've seen wants it middle center once they understand how it actually functions (put a pin in the exact middle of the popup and then pin that to the exact middle of the screen). Long / wide content ends up off the screen every time.

We see this ticket in support several times a month if not more, setting that as a preset would likely increase the number of these tickets we see.

Ultimately the positioning functionality used there needs overhauling, but in the meantime.

They want it to appear like its in the middle of the screen overall, setting top position a bit higher can help with that, but its dependent on the height of the content & screen really. That applies whether they use Top Center or Middle Center.

Reply:
@danieliser , okay. We had two different reviews just in the last week that both mentioned negatively that the popup wasn't vertically centered where I had responded mention the location for middle center which is what I was on my sites that use PM. I didn't realize there were scenarios where that setting would be an issue.

I'll have to test out the top center plus top position as you suggested to get the preset updated.

Reply:
I'm ok with having that preset, but there should be some notes made somewhere to indicate the potential issue and try to alleviate the support requests that will ensue.

Alternatively we can look at tweaking the positioning methodology to never put the popup off screen to the top. IE add in a failsafe so the negative side of Middle Center doesn't occur.

Might be an easy fix where we just compare `top >= 0 ? top : 0;`.

Reply:
Honestly that is probably the smarter way to go. Leave your code as middle center and lets take a crack at the other issue while we are at it.